File: January Challenge/Java/198. House Robber.java
class Solution {
    public int rob(int[] nums) {
        int[] prev = new int[nums.length];
        prev[0] = nums[0];

        for (int i = 1; i < nums.length; i++) {
            int[] curr = new int[nums.length];

            int take = nums[i] + (i - 2 >= 0 ? prev[i - 2] : 0);
            int not_take = prev[i - 1];

            curr[i] = Math.max(take, not_take);
            prev[i] = curr[i];
        }

        return prev[nums.length - 1];
    }
}

File: January Challenge/Java/198. House Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        
        def helper():
            prev = [0 for _ in range(len(nums))]
            prev[0] = nums[0]

            for i in range(1, len(nums)):
                curr = [0 for _ in range(len(nums))]

                take = nums[i] + prev[i-2]
                not_take = 0 + prev[i-1]

                curr[i] = max(take, not_take)
                
                prev[i] = curr[i]
            return prev[len(nums)-1]

        return helper()
        
